Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -381,6 +381,45 @@ Needle contains a module aimed to bypass non-specific jailbreak detection implem
 [needle][script_jailbreak-detection-bypass] > run
 ```
 
+
+
+#### Using getppid
+
+Applications on iOS can detect if they have been started by a debugger by checking their parent PID. Normally, an application is started by [launchd](http://newosxbook.com/articles/Ch07.pdf) process, which is the first process running in the *user mode* and has PID=1. However, if a debugger starts an application, we can observe that `getppid()` returns PID different than 1. The whole detection can be implemented in the following way:
+
+```objective-c
+- (bool)AmIBeingDebugged{
+    int ppid = getppid();
+    return ppid != 1;
+}
+```
```

Comment:
What about directly the Swift version?
```suggestion
```swift
func AmIBeingDebugged() -> Bool {
    return getppid() != 1
}```
```

Reply:
Right, it's probably more convenient for developers to have Swift code